### PR TITLE
chore(test): remove global test scheduler 

### PR DESCRIPTION
**Pull request by tmair**
**Description:**
This PR removes the global `rxTestScheduler` from the test infrastructure. To get this working it was necessary to also remove the tests for the global test scheduler.

In addition the custom mocha UI was also removed because it is no loger necessary.

### CODE REVIEW

**Review by benlesh — APPROVED** [review]

**Comment by benlesh** [comment]
This is great, @tmair ... it looks like there's a conflict though.

File: spec/schedulers/TestScheduler-spec.ts [hunk 1]
```diff
@@ -1,15 +1,14 @@
+/** @prettier */
```
Comment (jakovljevic-mladen) [inline spec/schedulers/TestScheduler-spec.ts:1]: I hope this is the last PR where we have this line included. Can we update `.prettierrc.json` and completely remove `"overrides"` filed from it? 🤔

**Comment by benlesh** [comment]
Thank you so much, @tmair !

File: spec/schedulers/TestScheduler-spec.ts (hunk 1 above)
Comment (tmair) [reply spec/schedulers/TestScheduler-spec.ts:1]: @jakovljevic-mladen I created a draft PR that removes the pragma and the `spec` directory from the prettier formatting override. Please have a look at #7275. If we would like to merge those changes I can update the PR and promote it to a proper PR.

Comment (jakovljevic-mladen) [reply spec/schedulers/TestScheduler-spec.ts:1]: > If we would like to merge those changes I can update the PR and promote it to a proper PR.

I think we do. Great @tmair, thanks a bunch!

**Pull request merged**